textpipe/doc.py

```diff
@@ -512,21 +513,51 @@ def aggregate_word_vectors(self, model_name=None, aggregation='mean', normalize=
         else:
             raise NotImplementedError(f'Aggregation method {aggregation} is not implemented.')
 
-    def _load_gensim_word2vec_model(self, model_file=None):
+    def _load_gensim_word2vec_model(self,
+                                    model_file=None,
+                                    redis_host=None,
+                                    redis_port=None,
+                                    redis_db=0,
+                                    max_lru_cache_size=1024):
         """
-        Loads pre-trained Gensim word2vec keyed vector model
+        Loads pre-trained Gensim word2vec keyed vector model from either local or Redis
         >>> model = Doc('')._load_gensim_word2vec_model('tests/models/gensim_test_nl.kv')
         >>> type(model)
         <class 'gensim.models.keyedvectors.Word2VecKeyedVectors'>
         """
         lang = self.language if self.is_reliable_language else self.hint_language
         if not self._gensim_vectors or lang not in self._gensim_vectors:
-            vectors = KeyedVectors.load(model_file, mmap='r')
+            if redis_host and redis_port:
+                vectors = RedisKeyedVectors(redis_host,
+                                            redis_port,
+                                            redis_db,
+                                            lang,
+                                            max_lru_cache_size)
+                if not vectors.exists:
+                    raise TextpipeMissingModelException(f'Redis does not contain a model '
+                                                        f'for language {lang}. The model '
+                                                        f'needs to be loaded before use '
+                                                        f'(see load_keyed_vectors_into_redis).')
+            elif model_file:
+                try:
+                    vectors = KeyedVectors.load(model_file, mmap='r')
+                except FileNotFoundError:
+                    raise TextpipeMissingModelException(
+                        f'Gensim keyed vector file {model_file} is not available.')
+            else:
+                raise TextpipeMissingModelException(
+                    'Either specify model_file or redis information')
             self._gensim_vectors[lang] = vectors
         return self._gensim_vectors[lang]
 
     @functools.lru_cache()
-    def generate_gensim_document_embedding(self, model_file=None, lowercase=True):
+    def generate_gensim_document_embedding(self,
+                                           model_file=None,
+                                           lowercase=True,
+                                           redis_host=None,
```

Comment:
I am not very keen on having these Redis specific attributes in a public method in `Doc`. 

1. How about we change `model_file` to `model_uri`?
2. Would it make sense to have `RedisKeyedVectors` also work for the `file://` URI case? Then all the logic of using Redis could go there?

Reply:
`model_uri` would work. Does this imply that all filenames should be prefixed with `file://`? I would not like that. How about handling strings without scheme prefix as files, and strings conform the `redis://[host]:[port]/[db]` format as redis configs?

Reply:
@dodijk I implemented the solution I proposed above